src/GotHttpAdapter.ts

```diff
@@ -126,6 +128,13 @@ class GotHttpAdapter implements HttpAdapter {
       });
     }
 
+    if (err instanceof GotParseError) {
+      throw new ParseError({
+        message: err.message.split(/\sin\s+"?https?:\/{2}/i, 1)[0],
```

Comment:
Do we really need that? When would the message be something meaningful and not `Unexpected token at..`?

Reply:
It could be `Unknown body type...` like seen at https://github.com/sindresorhus/got/blob/efafa167d70bb68591cba89343d3fae9a69d8cb1/source/core/response.ts#L146

Reply:
It's either text or json, and it cannot throw for text if my logic from above is correct.

Reply:
So what's your suggestion?